flit_core/flit_core/sdist.py

```diff
@@ -144,8 +144,9 @@ def apply_includes_excludes(self, files):
         crucial_files = set(
             self.extra_files + [str(self.module.file.relative_to(self.cfgdir))]
         )
+
         missing_crucial = crucial_files - files
-        if missing_crucial:
+        if missing_crucial and not self.module.name.endswith('-stubs'):
```

Comment:
A check like this feels wrong. Ideally, `__init__.py` wouldn't even be added to `crucial_files` for stub packages. Furthermore, it might make sense to do the `endswith("-stubs")` check once and store it if it's a stubs package in the `flit.common.Module` so we can access it here, e.g. with `self.module.is_stubs_pkg`.

Reply:
> A check like this feels wrong. Ideally, __init__.py wouldn't even be added to crucial_files for stub packages.

that's why I skip the check here, if it's a stub package.

> Furthermore, it might make sense to do the endswith("-stubs") check once and store it if it's a stubs package in the flit.common.Module so we can access it here, e.g. with self.module.is_stubs_pkg.

make sence